packages/domain-picker/src/domain-picker/index.tsx

```diff
@@ -94,21 +77,28 @@ const DomainPicker: FunctionComponent< Props > = ( {
 	onMoreOptions,
 	quantity = PAID_DOMAINS_TO_SHOW,
 	currentDomain,
-	domainSearch,
-	onSetDomainSearch,
-	domainCategory,
-	onSetDomainCategory,
-	domainSuggestions,
 	analyticsFlowId,
 	analyticsUiAlgo,
-	domainCategories,
 	domainSuggestionVendor,
+	initialDomainSearch,
+	onSetDomainSearch,
 } ) => {
 	const { __ } = useI18n();
 	const label = __( 'Search for a domain' );
 
 	const [ currentSelection, setCurrentSelection ] = useState( currentDomain );
 
+	// keep domain query in local state to allow free editing of the input value while the modal is open
+	const [ domainSearch, setDomainSearch ] = useState< string >( initialDomainSearch );
+	const [ domainCategory, setDomainCategory ] = useState< string | undefined >();
+
+	const domainSuggestions = useDomainSuggestions(
+		domainSearch.trim(),
+		domainCategory,
+		useI18n().i18nLocale,
+		10
```

Comment:
```suggestion
		quantity + 1
```
This will fix the jump mentioned in https://github.com/Automattic/wp-calypso/pull/43045/files#r437401116